lsp/src/server/utils/http/request.ts

```diff
@@ -0,0 +1,34 @@
+
+import { Headers, XHROptions, XHRRequest, XHRResponse, xhr } from 'request-light'
+
+
+
+// Abstract away interfaces of existing implementations
+export type HttpRequestOptions = Pick<XHROptions, 'headers'>
+export type HttpRequestHeaders = Headers
+export type HttpResponse = XHRResponse
+
+export interface HttpRequesterI {
+    request(url: string, options?: HttpRequestOptions): Promise<HttpResponse>
+}
+/**
+ * Abstracted http request class.
+ * 
+ * This will ensure we are not locked in to a specific
+ * http request implementation.
+ */
+export class HttpRequester implements HttpRequesterI {
```

Comment:
Modularity/decoupling is great and all but it can make things more complicated, especially when the decoupling is unnecessary. Is there any reason why this `HttpRequester` class couldn't be implemented later when (if) we did have multiple libraries? What's wrong with just using `request-light` directly?